typing/printtyp.ml

```diff
@@ -1842,37 +1803,68 @@ let explanation env unif t3 t4 : (Format.formatter -> unit) option =
         fprintf ppf
           "@,@[The %s object type has an abstract row, it cannot be closed@]"
           (if t4.desc = Tnil then "first" else "second"))
```

Comment:
Why are the three cases above still here?
Would their disappearance require cleaning up typeclass.ml?

Reply:
It seems that emitting the right information in Ctype was enough.